app/src/main/java/app/grapheneos/pdfviewer/fragment/PasswordPromptFragment.kt

```diff
@@ -0,0 +1,87 @@
+package app.grapheneos.pdfviewer.fragment
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextUtils
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.inputmethod.InputMethodManager
+import android.widget.Button
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import app.grapheneos.pdfviewer.PdfViewer
+import app.grapheneos.pdfviewer.R
+import app.grapheneos.pdfviewer.databinding.PasswordDialogFragmentBinding
+import com.google.android.material.textfield.TextInputEditText
+
+class PasswordPromptFragment : DialogFragment() {
+
+    private lateinit var passwordDialogFragmentBinding: PasswordDialogFragmentBinding
+    private lateinit var passwordEditText : TextInputEditText
+    private val imm by lazy { requireContext().getSystemService(InputMethodManager::class.java) }
+
+    private fun enableDisablePositiveButton(enable: Boolean) {
+        val positiveButton: Button =
+            (dialog as AlertDialog).getButton(DialogInterface.BUTTON_POSITIVE)
+        println("enableDisablePositiveButton positiveButton $positiveButton enable $enable ")
+        if (enable) {
+            if (!positiveButton.isEnabled) {
+                positiveButton.isEnabled = true
+            }
+        } else if (positiveButton.isEnabled) {
+            positiveButton.isEnabled = false
+        }
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val passwordPrompt = AlertDialog.Builder(requireContext())
+        passwordDialogFragmentBinding =
+            PasswordDialogFragmentBinding.inflate(LayoutInflater.from(requireContext()))
+        passwordEditText = passwordDialogFragmentBinding.pdfPasswordEdittext
+        passwordPrompt.setView(passwordDialogFragmentBinding.root)
+        passwordEditText.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
+            override fun onTextChanged(input: CharSequence, i: Int, i1: Int, i2: Int) {
+                enableDisablePositiveButton(input.isNotEmpty())
+            }
+
+            override fun afterTextChanged(editable: Editable) {}
+        })
+        passwordPrompt.setPositiveButton(
+            R.string.ok
+        ) { _, _ ->
+            dismissSoftInput()
+            val password = passwordEditText.text.toString()
+            (activity as PdfViewer).loadPdfWithPassword(password)
+            dismiss()
+        }
+        passwordPrompt.setNegativeButton(
+            R.string.no
+        ) { _, _ ->
+            dismissSoftInput()
+            dismiss()
+        }
+        val dialog = passwordPrompt.create()
+        isCancelable = false
```

Comment:
`isCancelable` -> `isCancellable`

Reply:
this is not something i have defined rather it's [system/framework api](https://developer.android.com/reference/androidx/fragment/app/DialogFragment#setCancelable(boolean))